### ISSUE TITLE
Review and complete model training setup code

### ISSUE THREAD

**Issue by koann0107** [prody/Pharmmaker#19]
The current model setup code needs review and completion for proper GPU training implementation. Here are the key points that need to be addressed:

### Current Implementation
```python
import segmentation_models_pytorch as smp
import monai
import torch
import time
from datetime import datetime

unet = monai.networks.nets.UNet(
    spatial_dims=3,
    in_channels=1,
    out_channels=2,
    channels=(8, 16, 32, 64),
    strides=(2, 2, 2),
)

model = Model(
    net=unet,
    criterion=monai.losses.DiceLoss(softmax=True),
    learning_rate=1e-2,
    optimizer_class=torch.optim.AdamW,
)
early_stopping = pl.callbacks.early_stopping.EarlyStopping(
    monitor='val_loss',
)
if torch.cuda.is_available():
    precision = 16
else:
    precision = 32
trainer = pl.Trainer(
    devices=1,
    precision=precision,
    callbacks=[early_stopping],
    max_epochs=10,
    log_every_n_steps=1,
)
trainer.logger._default_hp_metric = False
```

### Issues to Address

1. **Missing Imports and Dependencies**:
   - `pytorch_lightning` (pl) import is missing despite using its components
   - `Model` class is used but not defined or imported

2. **Data Loading**:
   - No data loading or dataset implementation is shown
   - Need to implement DataLoader and Dataset classes
   - Missing data preprocessing and augmentation steps

3. **GPU Configuration**:
   - While `devices=1` is set and CUDA availability is checked, explicit device moving for model and data is not shown
   - Consider adding `accelerator='gpu'` to the Trainer configuration

4. **Model Class Definition**:
   - The `Model` class needs to be properly defined, likely inheriting from `pl.LightningModule`
   - Need to implement required methods like `training_step`, `validation_step`, etc.

### Recommendations

1. Add missing imports:
```python
import pytorch_lightning as pl
```

2. Implement proper Model class:
```python
class Model(pl.LightningModule):
    def __init__(self, net, criterion, learning_rate, optimizer_class):
        super().__init__()
        self.net = net
        self.criterion = criterion
        self.learning_rate = learning_rate
        self.optimizer_class = optimizer_class

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self.net(x)
        loss = self.criterion(y_hat, y)
        self.log('train_loss', loss)
        return loss

    def validation_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self.net(x)
        loss = self.criterion(y_hat, y)
        self.log('val_loss', loss)
        return loss

    def configure_optimizers(self):
        return self.optimizer_class(self.parameters(), lr=self.learning_rate)
```

3. Update Trainer configuration:
```python
trainer = pl.Trainer(
    accelerator='gpu',
    devices=1,
    precision=precision,
    callbacks=[early_stopping],
    max_epochs=10,
    log_every_n_steps=1,
)
```

4. Implement data loading:
```python
# Example DataLoader implementation
train_dataloader = DataLoader(
    train_dataset,
    batch_size=batch_size,
    shuffle=True,
    num_workers=num_workers
)

val_dataloader = DataLoader(
    val_dataset,
    batch_size=batch_size,
    shuffle=False,
    num_workers=num_workers
)
```

5. Add training execution:
```python
trainer.fit(model, train_dataloader, val_dataloader)
```

### Additional Considerations
- Consider adding learning rate scheduling
- Implement proper logging and model checkpointing
- Add proper data validation and error handling
- Consider implementing testing phase
- Add proper documentation for model parameters and training configuration
